### ISSUE TITLE
False positive: exhaustive `match` over `bool` tuple wrongly reported as implicitly returning `None`

### ISSUE THREAD

**Issue by kamalfarahani** [astral-sh/ty#3807]
`ty` reports that the following function can implicitly return `None`, but the `match` statement is exhaustive over all possible `(bool, bool)` combinations, so every code path returns an `int`.

## Reproduction

```python
def f(b_1: bool, b_2: bool) -> int:
    match (b_1, b_2):
        case (True, True):
            return 0
        case (False, _):
            return 1
        case (_, False):
            return 2
```

## Diagnostic

```
Function can implicitly return `None`, which is not assignable to return type `int`
```

## Environment

- `ty` version: `0.0.50`

### Version

0.0.50
